utils/server/schemaPages.ts

```diff
@@ -0,0 +1,76 @@
+import { GetServerSidePropsContext } from "next";
+
+import prisma from "utils/server/prisma";
+import { getCachedSession } from "utils/server/session";
+import { SchemaPageHeaderProps } from "components/SchemaPageFrame/SchemaPageFrame";
+
+export const getSchemaPageHeaderData = async (
+	profileSlug: string,
+	contentSlug: string
+): Promise<SchemaPageHeaderProps | null> => {
+	const schema = await prisma.schema.findFirst({
+		where: {
+			slug: contentSlug,
+			agent: {
+				OR: [{ user: { slug: profileSlug } }, { organization: { slug: profileSlug } }],
+			},
+		},
+		select: {
+			id: true,
+			slug: true,
+			description: true,
+			isPublic: true,
+			agent: { select: { userId: true } },
+			updatedAt: true,
+		},
+	});
+
+	if (!schema) {
+		return null;
+	}
+
+	const versionCount = await prisma.schemaVersion.count({ where: { schemaId: schema?.id } });
+	const currentVersionDetails = await prisma.schemaVersion.findFirst({
+		where: {
+			schemaId: schema.id,
+		},
+		orderBy: { createdAt: "desc" },
+		select: {
+			versionNumber: true,
+			createdAt: true,
+		},
+	});
+	return {
+		contentSlug,
+		profileSlug,
+		versionCount,
+		currentVersionDate: currentVersionDetails!.createdAt.toISOString(),
+		currentVersionNumber: currentVersionDetails!.versionNumber,
+		schema: { ...schema, updatedAt: schema.updatedAt.toISOString() },
+	};
+};
+
+export const getSchemaPagePermissions = (
```

Comment:
I think this function is a good idea but it needs to be able to reject permissions on the new version and settings pages of public schemas for user who don't have access